src/hotspot/share/opto/chaitin.cpp

```diff
@@ -1637,32 +1653,40 @@ uint PhaseChaitin::Select( ) {
     }
 
     // Check if a color is available and if so pick the color
-    OptoReg::Name reg = choose_color( *lrg, chunk );
+    OptoReg::Name reg = choose_color(*lrg);
 
     //---------------
     // If we fail to color and the infinite flag is set, trigger
     // a chunk-rollover event
-    if (!OptoReg::is_valid(OptoReg::add(reg, -chunk)) && is_infinite_stack) {
+    if (!OptoReg::is_valid(reg) && is_infinite_stack) {
       // Bump register mask up to next stack chunk
-      chunk += RegMask::CHUNK_SIZE;
-      lrg->Set_All();
+      bool success = lrg->rollover();
```

Comment:
Can you add a comment that explains what this does / means? Do we start spilling to the stack slots instead of using registers?